docs/api-reference/browser_storage.md

```diff
@@ -42,73 +31,174 @@ Remove a cookie from the client's browser.
+which provides simple serialization helpers and works recursively in complex object structures.
+
+```python exec
+import inspect
+
+import reflex as rx
+
+from pcweb.base_state import State
+from pcweb.templates.docpage import docdemo
+
+
+class AppSettings(rx.Base):
+    theme: str = 'light'
+    sidebar_visible: bool = True
+    update_frequency: int = 60
+    error_messages: list[str] = []
+
+
+class ComplexLocalStorageState(State):
+    data_raw: str = rx.LocalStorage()
+    data: AppSettings = AppSettings()
+    settings_open: bool = False
+
+    def save_settings(self):
+        self.data_raw = self.data.json()
+        self.settings_open = False
+
+    def open_settings(self):
+        self.data = AppSettings.parse_raw(self.data_raw)
+        self.settings_open = True
+
+    def set_field(self, field, value):
+        setattr(self.data, field, value)
+
+
+def app_settings():
+    return rx.form(
+        rx.foreach(
+            ComplexLocalStorageState.data.error_messages,
+            rx.text,
+        ),
+        rx.form_label(
+            "Theme",
+            rx.input(
+                value=ComplexLocalStorageState.data.theme,
+                on_change=lambda v: ComplexLocalStorageState.set_field("theme", v),
+            ),
+        ),
+        rx.form_label(
+            "Sidebar Visible",
+            rx.switch(
+                is_checked=ComplexLocalStorageState.data.sidebar_visible,
+                on_change=lambda v: ComplexLocalStorageState.set_field(
+                    "sidebar_visible",
+                    v,
+                ),
+            ),
+        ),
+        rx.form_label(
+            "Update Frequency (seconds)",
+            rx.number_input(
+                value=ComplexLocalStorageState.data.update_frequency,
+                on_change=lambda v: ComplexLocalStorageState.set_field(
+                    "update_frequency",
+                    v,
+                ),
+            ),
+        ),
+        rx.button("Save", type_="submit"),
+        on_submit=lambda _: ComplexLocalStorageState.save_settings(),
+    )
+
+def app_settings_example():
+    return rx.fragment(
+        rx.modal(
+            rx.modal_overlay(
+                rx.modal_content(
+                    rx.modal_header("App Settings"),
+                    rx.modal_body(app_settings()),
+                ),
+            ),
+            is_open=ComplexLocalStorageState.settings_open,
+            on_close=ComplexLocalStorageState.set_settings_open(False),
+        ),
+        rx.button("App Settings", on_click=ComplexLocalStorageState.open_settings),
+    )
+```
+
+```python eval
+docdemo(
+    inspect.getsource(AppSettings) + "\n\n" + inspect.getsource(
+        ComplexLocalStorageState,
+    ).replace("(State)", "(rx.State)"),
+    inspect.getsource(app_settings) + "\n\n" + inspect.getsource(
+        app_settings_example,
+    ).partition("return ")[2],
```

Comment:
These params are flipped around. The component code should be given first, then the state code